yt/frontends/ramses/tests/test_outputs.py

```diff
@@ -13,589 +14,494 @@
-    assert ("gas", "heating_compton") in ds.derived_field_list
-    assert ("gas", "cooling_primordial_prime") in ds.derived_field_list
-    assert ("gas", "cooling_compton_prime") in ds.derived_field_list
-    assert ("gas", "cooling_metal_prime") in ds.derived_field_list
-    assert ("gas", "heating_primordial_prime") in ds.derived_field_list
-    assert ("gas", "heating_compton_prime") in ds.derived_field_list
-    assert ("gas", "mu") in ds.derived_field_list
-
-    # Abundances
-    assert ("gas", "Electron_number_density") in ds.derived_field_list
-    assert ("gas", "HI_number_density") in ds.derived_field_list
-    assert ("gas", "HII_number_density") in ds.derived_field_list
-    assert ("gas", "HeI_number_density") in ds.derived_field_list
-    assert ("gas", "HeII_number_density") in ds.derived_field_list
-    assert ("gas", "HeIII_number_density") in ds.derived_field_list
-
-    def check_unit(array, unit):
-        assert str(array.in_cgs().units) == unit
-
-    check_unit(ds.r[("gas", "cooling_total")], "cm**5*g/s**3")
-    check_unit(ds.r[("gas", "cooling_primordial_prime")], "cm**5*g/(K*s**3)")
-    check_unit(ds.r[("gas", "number_density")], "cm**(-3)")
-    check_unit(ds.r[("gas", "mu")], "dimensionless")
-    check_unit(ds.r[("gas", "Electron_number_density")], "cm**(-3)")
-
-
-@requires_file(ramses_rt)
-def test_ramses_mixed_files():
-    # Test that one can use derived fields that depend on different
-    # files (here hydro and rt files)
-    ds = yt.load(ramses_rt)
-
-    def _mixed_field(field, data):
-        return data["rt", "photon_density_1"] / data["gas", "H_nuclei_density"]
-
-    ds.add_field(("gas", "mixed_files"), function=_mixed_field, sampling_type="cell")
-
-    # Access the field
-    ds.r[("gas", "mixed_files")]
-
-
 ramses_empty_record = "ramses_empty_record/output_00003/info_00003.txt"
+ramses_mhd_128 = "ramses_mhd_128/output_00027/info_00027.txt"
 
 
-@requires_file(ramses_empty_record)
-def test_ramses_empty_record():
-    # Test that yt can load datasets with empty records
-    ds = yt.load(ramses_empty_record)
-
-    # This should not fail
-    ds.index
-
-    # Access some field
-    ds.r[("gas", "density")]
-
-
-@requires_file(ramses_new_format)
-@requires_module("f90nml")
-def test_namelist_reading():
-    ds = data_dir_load(ramses_new_format)
-    namelist_fname = os.path.join(ds.directory, "namelist.txt")
-    with open(namelist_fname, "r") as f:
-        ref = f90nml.read(f)
-
-    nml = ds.parameters["namelist"]
-
-    assert nml == ref
-
-
-@requires_file(ramses_empty_record)
-@requires_file(output_00080)
-@requires_module("f90nml")
-def test_namelist_reading_should_not_fail():
-
-    for ds_name in (ramses_empty_record, output_00080):
-        # Test that the reading does not fail for malformed namelist.txt files
-        ds = data_dir_load(ds_name)
-        ds.index  # should work
-
-
-ramses_mhd_128 = "ramses_mhd_128/output_00027/info_00027.txt"
+@pytest.mark.answer_test
```

Comment:
I believe only the first test of this test suite is an answer test. The other are unit tests.

Reply:
In Jared's defense, I also blindly assume that `frontends/<x>/tests/test_outputs.py` is dedicated to answer tests. I don't know if we have an actual rule for this.